middlewares/logPageView.js

```diff
@@ -8,20 +8,19 @@ var sql = sqlLoader.loadSqlEquiv(__filename);
 
 module.exports = function(pageType) {
     return function(req, res, next) {
-        // Return immediately to keep processing the request. We are
-        // only going to log, so nothing here will be used to generate
-        // the response.
-        next();
-        
-        if (req.method != 'GET') return;
-        
-        if (!res.locals.user) return;
-        if (!res.locals.authn_user) return;
+        if (req.method != 'GET' || !res.locals.user || !res.locals.authn_user) {
+            next();
+            return;
+        }
 
         const user_id = res.locals.user ? res.locals.user.user_id : res.locals.authn_user.user_id;
-        // only log if we are the user who owns the viewed page
-        if (res.locals.instance_user && res.locals.instance_user.user_id != user_id) return;
-        
+
+        // Originally, we opted to only record page views for assessments if
+        // the authn'ed user is also the owner of the assessment instance.
+        // However, we now track all page views, so be sure to filter by
+        // authn_user_id if you only want page views from the student taking
+        // the assessment.
```

Comment:
I think we'll need such a filter in `instructorAssessmentInstance.sql` in the `BLOCK select_log`?

Reply:
It looks like that's already being done: https://github.com/PrairieLearn/PrairieLearn/blob/master/pages/instructorAssessmentInstance/instructorAssessmentInstance.sql#L308 (from master)

Reply:
Isn't that that JOIN just getting the user corresponding to the authn_user_id in the page_view_log? I don't see where it's actually checking that the authn_user_id is the same as the user who owns the assessment? Maybe something like a WHERE clause on `pvl.authn_user_id = ai.user_id`?

Reply:
Gah, you're absolutely right :) not sure what I was thinking there.